### ISSUE TITLE
Mismatch of dtypes when using `pow` and `quantile` expressions

### ISSUE THREAD

**Issue by philss** [pola-rs/polars#4796]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of polars.

### Issue Description

First of all, thank you for this project! :heart: 

While using [Expressions](https://pola-rs.github.io/polars-book/user-guide/dsl/expressions.html) I found a small inconsistency for [`pow`](https://pola-rs.github.io/polars/polars/prelude/enum.Expr.html#method.pow) and [`quantile`](https://pola-rs.github.io/polars/polars/prelude/enum.Expr.html#method.quantile) operations: when they are used in a Series of _dtype_ **integer**, they always return a Series of _dtype_ **float**.

I would like to know if this is expected, and if the user should manually cast back to integer, or if it is a bug.

### Reproducible Example

```rust
use polars::prelude::*;

fn main() {
    let df = df! [
        "my_int" => [Some(1), Some(2), Some(3), None, Some(5)],
    ]
    .unwrap();

    println!("{}", &df);

    let out = df
        .clone()
        .lazy()
        .select([
            col("my_int"),
            col("my_int").pow(2).alias("pow_2_my_int"),
            col("my_int")
                .quantile(0.3, QuantileInterpolOptions::Nearest)
                .alias("quantile_my_int"),
        ])
        .collect()
        .unwrap();

    println!("{}", out);
}
```

You can find this example at: https://github.com/philss/polars_mismatch_dtypes_example_app

### Expected Behavior

The expressions `pow` and `quantile` should respect the original _dtype_ of the series.
So if you run `col("my_int").pow(2)`, the resultant series should be of type integer, unless the exponent is a float.

Same for `quantile`: using it in a series of type **integer** should return a series of type **integer**.

### Installed Versions

<details>

* Features: "fmt", "lazy", "performant", "rows", "simd"

* Installed version: 0.23.2

* Rust version: rustc 1.64.0-nightly (93ffde6f0 2022-07-23)

* OS: Linux with Kernel version 5.15.0-47-generic.

PS: this problem was found in https://github.com/elixir-nx/explorer/pull/339
</details>

**Comment by slonik-az** [comment]
Actually, `pow` applied to two integers can easily result in a float. For example:  $2^{-1} = \frac{1}{2}$. So, it is perfectly fine that the return type of `pow` becomes `float`. Algebraically, integers do not form a field and quite a few operations on integers result in fractions or reals (pow, div).

The same is true for quantile function. It is an inverse of `cdf` (Cummulative Distribution Funciton) that also better expressed with real numbers, hence floats. There might be some special cases where integers would be sufficient, but it would be corner cases better avoided. 

Bottom line, input-output type consistency is great but when it collides with mathematical correctness, Mathematics wins 😄

**Comment by philss** [comment]
@slonik-az got it! Thank you for the explanation :smile: 

So float will be :)
